src/nav-bar/NavBar.js

```diff
@@ -37,63 +50,81 @@ const NavList = styled.ul`
         }
       }
     }
+
+    /* List Item Link Style Overrides */
+
+    .branding-bar {
+      font-size: 24px;
+      &--active {
+        &::after {
+          content: "";
+          position: absolute;
+          left: calc(100% + 16px);
+          top: 50%;
+          width: 56px;
+          border-top: 2px solid ${(props) => props.theme.colors.highlight};
+        }
+      }
+    }
+
+    .overview {
+      border-top: 1px solid ${(props) => props.theme.colors.divider};
+      font-size: 64px;
+      width: 100%;
+      ${fluidFontSizeStyles(32, 64)}
+    }
+
+    .secondary {
+      &--active {
+        border-bottom: 2px solid ${(props) => props.theme.colors.highlight};
+
+        &::after {
+          content: none;
+        }
+      }
+    }
   }
 `;
 
-const addLinkClasses = ({ isCurrent }) => ({
-  className: classNames("NavItem__link", {
-    "NavItem__link--active": isCurrent,
-  }),
+const addLinkClasses = (overrideClass, isCurrent) => ({
+  className: classNames(
+    "NavItem__link",
+    overrideClass,
+    { "NavItem__link--active": isCurrent },
+    { [`${overrideClass}--active`]: isCurrent }
+  ),
 });
 
-export default function NavBar({ large, onClick }) {
+export default function NavBar({ pages, onClick, className, nested }) {
```

Comment:
Alright, in this [latest commit](https://github.com/Recidiviz/public-dashboard/pull/60/commits/4409b22681bb77950d670122783584f1f9715f7c) I was able to get all of the styles and their overrides defined solely in this `NavBar` component.  It still isn't pretty, but overall feels more correct, and if nothing else, limits the messiness to one spot.

And this definitely feels a whole lot better:

```html
<NavBar className="secondary" />
```